sycl/plugins/cuda/pi_cuda.hpp

```diff
@@ -591,42 +596,39 @@ struct _pi_kernel {
   struct arguments {
     static constexpr size_t MAX_PARAM_BYTES = 4000u;
     using args_t = std::array<char, MAX_PARAM_BYTES>;
-    using args_size_t = std::vector<size_t>;
-    using args_index_t = std::vector<void *>;
+    using args_size_t = std::array<size_t, MAX_PARAM_BYTES>;
+    using args_index_t = std::array<void *, MAX_PARAM_BYTES>;
     args_t storage_;
     args_size_t paramSizes_;
     args_index_t indices_;
```

Comment:
```suggestion
    args_index_t indices_ = { implicitOffsetArgs_ };
```

Reply:
Shouldn't this be `= { &implicitOffsetArgs_ }`?